src/include/optimizer/optimizer_metadata.h

```diff
@@ -19,6 +19,8 @@
 #include "optimizer/rule.h"
 #include "settings/settings_manager.h"
 
+#include <memory>
+
```

Comment:
Some documentation of what OptimizerMetadata contains in optimizer_metadata would be nice as well.